spec/graphql/types/work_type_spec.rb

```diff
@@ -1201,4 +1201,93 @@
       )
     end
   end
+
+  describe "query works with repository subjects" do
+    before :all do
+      SLEEP_TIME = 2
+      WORK_COUNT = 10
+
+      DataciteDoi.import(force: true)
+      Client.import(force: true)
+      Prefix.import(force: true)
+      ClientPrefix.import(force: true)
+      ReferenceRepository.import(force: true)
+      Event.import(force: true)
+
+      search_query = '
+        fragment facetFields on Facet {
+          id
+          title
+          count
+        }
+        query{
+          works(query:"*"){
+            totalCount
+            fieldsOfScience { ...facetFields }
+            fieldsOfScienceRepository { ...facetFields }
+            fieldsOfScienceCombined{ ...facetFields }
+          }
+        }
+      '
+
+      create(:prefix)
+      client = create(:client_with_fos)
+      create_list(:doi, WORK_COUNT,
+        aasm_state: "findable",
+        client: client
+      )
+      Doi.import
+      sleep SLEEP_TIME
+      @facet_response = LupoSchema.execute(search_query).as_json
+      Rails.logger.level = :fatal
+      DataciteDoi.destroy_all
+      ReferenceRepository.destroy_all
+      Client.destroy_all
+      Provider.destroy_all
+      Prefix.destroy_all
+      ClientPrefix.destroy_all
+      ProviderPrefix.destroy_all
+      Event.destroy_all
+    end
+
+    after :all do
+    end
+
+    it "has all dois in the search results" do
+      response = @facet_response
+      expect(response.dig("data", "works", "totalCount")).to eq(WORK_COUNT)
+    end
+
+    it "returns Field of Science Facets" do
+      response = @facet_response
+      expect(
+        response.dig("data", "works", "fieldsOfScience")
+      ).to match_array([])
+    end
+
+    it "returns Field of Science Facets from the repository" do
+      response = @facet_response
+      expect(
+        response.dig("data", "works", "fieldsOfScienceRepository")
+      ).to match_array([
+        {
+          "id" => "example_subject",
+          "title" => "Example Subject",
+          "count" => WORK_COUNT
+        }
+      ])
+    end
+    it "returns combined Field of Science Facets" do
+      response = @facet_response
+      expect(
+        response.dig("data", "works", "fieldsOfScienceCombined")
+      ).to match_array([
+        {
+          "id" => "example_subject",
+          "title" => "Example Subject",
+          "count" => WORK_COUNT
+        }
+      ])
```

Comment:
I might be getting the wrong end of the stick - but it looks like the data being tested here only comes from one half of the combined source (the repository). Would it be worth adding another subject record to the work, so that there is one coming from either part of the combination, and testing against that to ensure that the data is getting combined correctly?